### PR TITLE
Generate and distribute trust quorum shares during rack setup

### LINKED ISSUE #513: Remove key-share file reading when trust-quorum keys are dynamically generated
In #487 we read from key share files provisioned by a deployment system (or manually user generated). If a key share file is not present, we operate in single-node mode. 

This issue is here to remind us to remove the key share file reading code from the bootstrap agent when we start dynamically generating trust quorum keys during rack initialization. We will most likely replace this with a boolean option to indicate single node mode.

### PR DESCRIPTION
**Pull request by jgallagher**
The primary change in this PR is that trust quorum shares are now created by RSS and sent to sleds as part of the `SledAgentRequest`s distributed during rack setup instead of being generated ahead of time and installed by `thing-flinger`. Ancillary changes:

* Share retrieval during rack unlock now happens under sprockets (via the bootstrap-agent server introduced in #1173)
* Removes the SPDM-based trust quorum client/server
* Removes SPDM

Closes #513 (complete) and #517 (no longer relevant).

### CODE REVIEW

**Comment by smklein** [comment]
image

Love to see this!

File: sled-agent/src/rack_setup/service.rs [hunk 1]
```diff
@@ -264,8 +268,42 @@ impl ServiceInner {
...
+                        "Skipping rack secret creation due to config",
+                        " (despite discovery of {} bootstrap agents)"
+                    ),
+                    total_shares,
+                );
+                None
+            }
+        } else {
+            None
+        };
```
Comment (smklein) [inline sled-agent/src/rack_setup/service.rs:305]: WDYT about splitting this into a separate function so we can add a test?

**Review by smklein — APPROVED** [review]

File: sled-agent/src/rack_setup/service.rs (hunk 1 above)
Comment (jgallagher) [reply sled-agent/src/rack_setup/service.rs:305]: Seems perfectly reasonable; will do.

**Review by andrewjstone — APPROVED** [review]
🚀 Bye-Bye SPDM!

**Pull request merged**